main.cpp

```diff
@@ -1,479 +1,535 @@
-        a) For each plain-english UDT, write out 5 traits or 
-        properties and 3 things it can do.
-        b) remember: these properties will eventually become
-        primitives.
-        c) pick properties that can eventually be represented with 
-        'int float double bool char std::string'.
-*/
 
 /*
-Thing 1)
+Thing 1) Coffee shop
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) Number of employees (int)
+    2) Number of coffee maker machines (int)
+    3) Open/closed state (bool)
+    4) Name (std::string)
+    5) Date established (int)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) Sell coffee
+    2) Purchase coffee beans
+    3) Toggle open/closed state
  */
 
+struct CoffeeShop
+{
+    // number of employees
+    int numEmployees = 10;
+    // number of coffee maker machines
+    int numCoffeeMachines = 2;
+    // open/closed state boolean
+    bool isOpen = true;
+    // name
+    std::string name = "The Nautical Bean";
+    // date established
+    int dateEstablished = 1995;
+
+    struct Employee
+    {
+        std::string firstName = "Jane";
+        std::string surname = "Doe";
+        int age = 20;
+        std::string role = "Cashier";
+        int id = 0;
+
+        void setRole(std::string newRole = "Generalist");
+        void setSurname(std::string newSurname);
+        bool isOlderThan(int ageToCompare = 0);   // returns true if employee age exceeds ageToCompare
+    };
+
+    Employee employeeOnDuty;
+
+    Employee getEmployeeOnDuty();  // returns the employeeOnDuty
+    void changeName(std::string newName);
+    void setEmployeeOnDuty(Employee employee);
+    void open();
+    void close();
+};
+
 /*
-Thing 2)
+Thing 2) Dog
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) name (std::string)
+    2) age in months (int)
+    3) breed (std::string)
+    4) weight (float)
+    5) running speed (float)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) run
+    2) eat food
+    3) drink water
  */
 
+struct Dog
+{
+    // name
+    std::string name = "Watson";
+    // age in months
+    int ageInMonths = 26;
+    // breed
+    std::string breed = "Hound Mix";
+    // weight
+    float weightInPounds = 68.4f;
+    // running speed
+    float runningSpeed = 24.5f;
+
+    void run();
+    void eatFood();
+    void drinkWater();
+};
+
 /*
-Thing 3)
+Thing 3) Table
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) number of legs (int)
+    2) height (float)
+    3) tabletop radius (float)
+    4) color (std::string)
+    5) weight (float)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) hold items
+    2) collapse
+    3) fold closed
  */
 
+struct Table
+{
+    // number of legs
+    int numLegs = 4;
+    // height
+    float height = 39.f;
+    // tabletop radius
+    float tabletopRadius = 14.f;
+    // color
+    std::string color = "White";
+    // weight
+    float weight = 68.5f;
+
+    void supportItem(int itemWeight = 0);
```

Comment:
The plain-english description is `hold items`.
the function's name should match this
- [ ] revise